src/spikeinterface/sortingcomponents/waveforms/savgol_denoiser.py

```diff
@@ -7,22 +7,47 @@
 from spikeinterface.sortingcomponents.peak_pipeline import PipelineNode, WaveformExtractorNode
 
 
-class SavGolDenoiser(PipelineNode):
+class SavGolDenoiser(WaveformExtractorNode):
```

Comment:
Should this be propagated to the tPCA_denoiser and NNDenoiser?

Reply:
I think yes, all nodes that are expected to returns waveforms (and thus denoiser) should behave as WaveformExtractorNodes to inherit params from their parents (nbefore/nafter). This is why the modified init here and the search for a wavefor_extractor in the parents

Reply:
You want me to do that in the branch?

Reply:
Since you already added it to the `SavGolDenoiser` I think it makes sense to propagate it to all denoisers ;) yeah let's do it in this branch

Reply:
I did it for the neural network, but for the TemporalPCA, I think the person that wrote the code should do it. Just because it seems to me that there is some useless complexity that should be cleaned if we adopt such a scheme.

Reply:
Actually, I also did it for TemporalPCA

Reply:
Minor caveat... I deactivate one of the test of the TemporalPCA node for now, because I saw that there is, in the detect_peak refactoring, a function called find_parent() that will do the job I'm currently doing. We should reactivate the tests while using this function

Reply:
You can leave the tests deactivated if you want. Now that we have agreed on using a common base class for waveform nodes I will refactor and add a general test.